libvast/vast/table_slice.hpp

```diff
@@ -143,6 +143,21 @@ expected<std::vector<table_slice_ptr>>
 make_random_table_slices(size_t num_slices, size_t slice_size,
                          record_type layout, id offset = 0, size_t seed = 0);
 
+/// Selects all rows in `xs` with event IDs in `selection` and appends produced
+/// table slices to `result`. Cuts `xs` into multiple slices if `selection`
+/// produces gaps.
+/// @pre `xs != nullptr`
+void select(std::vector<table_slice_ptr>& result, const table_slice_ptr& xs,
+            const ids& selection);
```

Comment:
Looks like this function is an implementation detail. If so, it could go into the `cpp` file instead.

Otherwise, documenting the parameters via `@param` would be helpful.

Reply:
This is not an implementation detail, see commit message for a68a07c. I'll add the params.